js/views/conversation_search_view.js

```diff
@@ -1,167 +1,168 @@
+    initialize(options) {
+      this.$input = options.input;
+      this.$new_contact = this.$('.new-contact');
 
-        filterContacts: function() {
-            var query = this.$input.val().trim();
-            if (query.length) {
-                if (this.maybeNumber(query)) {
-                    this.new_contact_view.model.set('id', query);
-                    this.new_contact_view.render().$el.show();
-                    this.new_contact_view.validate();
-                    this.hideHints();
+      this.typeahead = new Whisper.ConversationCollection();
+      // View to display the matched contacts from typeahead
+      this.typeahead_view = new Whisper.ConversationListView({
+        collection: new Whisper.ConversationCollection([], {
+          comparator(m) { return m.getTitle().toLowerCase(); },
+        }),
+      });
+      this.$el.append(this.typeahead_view.el);
+      this.initNewContact();
+      // this.listenTo(this.collection, 'reset', this.filterContacts);
+      this.pending = Promise.resolve();
+    },
 
-                } else {
-                    this.new_contact_view.$el.hide();
-                }
-                this.pending = this.pending.then(function() {
-                    return this.typeahead.search(query).then(function() {
-                        this.typeahead_view.collection.reset(
-                            this.typeahead.filter(function(m) {
-                                return m.isSearchable();
-                            })
-                        );
-                    }.bind(this));
-                }.bind(this));
-                this.trigger('show');
-            } else {
-                this.resetTypeahead();
-            }
-        },
+    events: {
+      'click .new-contact': 'createConversation',
+    },
 
-        initNewContact: function() {
-            if (this.new_contact_view) {
-                this.new_contact_view.undelegateEvents();
-                this.new_contact_view.$el.hide();
-            }
-            // Creates a view to display a new contact
-            this.new_contact_view = new Whisper.NewContactView({
-                el: this.$new_contact,
-                model: ConversationController.createTemporary({
-                    type: 'private'
-                })
-            }).render();
-        },
+    filterContacts() {
+      const query = this.$input.val().trim();
+      if (query.length) {
+        if (this.maybeNumber(query)) {
+          this.new_contact_view.model.set('id', query);
+          this.new_contact_view.render().$el.show();
+          this.new_contact_view.validate();
+          this.hideHints();
+        } else {
+          this.new_contact_view.$el.hide();
+        }
+        // NOTE: Temporarily allow `then` until we convert the entire file
+        // to `async` / `await`:
+        /* eslint-disable more/no-then */
+        this.pending = this.pending.then(() => this.typeahead.search(query).then(() => {
```

Comment:
This formatting is not very readable, at least to me. Probably a newline for every `.then()` to make the chaining easy to follow?

Reply:
Agreed, fixed it:

```javascript
this.pending = this.pending.then(() =>
  this.typeahead.search(query).then(() => {
    this.typeahead_view.collection.reset(this.typeahead.filter(isSearchable));
  }));
```

Note that the nesting is strange because it’s not a linear `then` chain. It’s important we preserve formatting that ESLint autofix is happy with, otherwise we’ll lose the ability to use it.